src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -19,6 +19,23 @@ use once_cell::sync::Lazy;
 use regex::{Captures, Regex};
 use serde_json::json;
 
+use gray_matter::engine::YAML;
+use gray_matter::Matter;
+use serde::Deserialize;
+
+fn extract_frontmatter(content: &str) -> Option<&str> {
+    let start = content.find("---")? + 3; // Find the end of the first `---` + 3 to move past it
```

Comment:
Given that `---` can appear anywhere in a regular markdown content, I'd restrict this to the case when the file starts with `---`.